### DISCUSSION TITLE
Rich Text Editor

### DISCUSSION THREAD

**Discussion by selimeneserd** [shadcn-ui/ui#3185] [Blocks Request]
Controllable Rich Text Editor would be great like Mantine. Using TipTap is fine as dependency.

**Comment by birobirobiro** [comment]
Look this: https://platejs.org/

**Reply by selimeneserd** [reply]
Yeap, I know this. I am not expecting a RTE created from scratch. Just easier Tiptap based RTE to integrate to the project created using shadcn ui.

**Reply by amine-tayani** [reply]
yeah that would be great if it's added to shadcn ui.

**Reply by AbdullahMukadam** [reply]
> Yeap, I know this. I am not expecting a RTE created from scratch. Just easier Tiptap based RTE to integrate to the project created using shadcn ui.

Try this, it is same as mantine's rich text editor but for shadcn: https://www.rtecn.space/

**Comment by Salman2301** [comment]
This would be a nice feature! Just how the Mantine did, using tiptap

**Comment by htmujahid** [comment]
Here is one using shadcn and lexical [shadcn-editor.vercel.app](https://shadcn-editor.vercel.app)

**Comment by Mina-Massoud** [comment]
Hey everyone! 👋
I’ve just released a Rich Text Editor built with shadcn/ui and Tailwind CSS.

I made it because I was tired of complex or outdated editors — so here’s a clean, modern, and open-source alternative you can actually use in your projects.

👉 Live demo: [mina-rich-editor.vercel.app](https://mina-rich-editor.vercel.app/)

💻 GitHub repo: [github.com/Mina-Massoud/Mina-Rich-Editor](https://github.com/Mina-Massoud/Mina-Rich-Editor)

Would love to hear your feedback or suggestions!

**Comment by Prajwal-alphanimble** [comment]
you might wanna check out this discussion where I implemented a new resizable and dragable dashboard layout... this can be very helpfull to many developers in making analytic, sharable dashboard, the users can alter the dashboard based on their liking...

The disscussion link: https://github.com/shadcn-ui/ui/discussions/10466

**Comment by FlyingTurkman** [comment]
I have realesed RTE. Check it.

https://github.com/FlyingTurkman/rich-text-editor
